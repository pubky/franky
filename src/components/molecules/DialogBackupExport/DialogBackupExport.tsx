import Image from 'next/image';
import { QRCodeSVG } from 'qrcode.react';

import * as Atoms from '@/atoms';
import * as Config from '@/config';

interface DialogBackupExportProps {
  mnemonic?: string;
  children?: React.ReactNode;
}

const generateDeeplink = (mnemonicPhrase: string) => {
  const encodedMnemonic = encodeURIComponent(mnemonicPhrase);
  return `pubkyring://${encodedMnemonic}`;
};

const IMPORT_DESCRIPTION = (
  <>
    1. Open Pubky Ring, tap &apos;Add pubky&apos; <br />
    2. Choose &apos;Import pubky&apos; option <br />
    3. Tap &apos;Scan QR to import&apos; <br />
    4. Scan this QR code to import your recovery phrase
  </>
);

const EXPORT_DESCRIPTION = (
  <>
    1. Open Pubky Ring, tap &apos;Add pubky&apos; <br />
    2. Choose the &apos;Import pubky&apos; option
    <br />
    3. Scan this QR to import
  </>
);

export function DialogBackupExport({ mnemonic, children }: DialogBackupExportProps) {
  const qrValue = mnemonic ? generateDeeplink(mnemonic) : Config.PUBKY_CORE_URL;
  const descriptionContent = mnemonic ? IMPORT_DESCRIPTION : EXPORT_DESCRIPTION;

  return (
    <Atoms.Dialog>
      {children ? (
        <Atoms.DialogTrigger asChild>{children}</Atoms.DialogTrigger>
      ) : (
        <Atoms.DialogTrigger asChild>
          <Atoms.Button className="gap-2">
            <span>Continue</span>
          </Atoms.Button>
        </Atoms.DialogTrigger>
      )}
      <Atoms.DialogContent className="max-w-md" hiddenTitle={mnemonic ? 'Import recovery phrase' : 'Pubky Ring export'}>
        <Atoms.DialogHeader>
          <Atoms.DialogTitle>{mnemonic ? 'Import recovery phrase' : 'Pubky Ring export'}</Atoms.DialogTitle>
          <Atoms.DialogDescription>{descriptionContent}</Atoms.DialogDescription>
        </Atoms.DialogHeader>

        <Atoms.Container className="justify-between items-center">
          <Atoms.Container className="flex-row gap-6">
            <Atoms.Container>
              <Image src="/images/pubky-ring-phone.png" alt="App preview" width={250} height={430} />
            </Atoms.Container>

            <Atoms.Container className="gap-6">
              <Atoms.Container className="mx-0 bg-foreground rounded-lg p-4 w-[192px] h-[192px] items-center">
                <QRCodeSVG value={qrValue} size={192} />
              </Atoms.Container>
              <Atoms.Container className="gap-4">
                <Atoms.Link href={Config.APP_STORE_URL} target="_blank">
                  <Image src="/images/badge-apple.png" alt="App Store" width={120} height={40} />
                </Atoms.Link>
                <Atoms.Link href={Config.PLAY_STORE_URL} target="_blank">
                  <Image src="/images/badge-android.png" alt="Google Play" width={135} height={40} />
                </Atoms.Link>
              </Atoms.Container>
            </Atoms.Container>
          </Atoms.Container>
        </Atoms.Container>
      </Atoms.DialogContent>
    </Atoms.Dialog>
  );
}
