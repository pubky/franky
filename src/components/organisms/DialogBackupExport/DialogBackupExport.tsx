'use client';

import Image from 'next/image';
import { QRCodeSVG } from 'qrcode.react';

import * as Atoms from '@/atoms';
import * as Config from '@/config';
import * as Libs from '@/libs';

interface DialogBackupExportProps {
  mnemonic?: string;
  children?: React.ReactNode;
}

const IMPORT_DESCRIPTION = (
  <>
    1. Open Pubky Ring, tap &apos;Add pubky&apos; <br />
    2. Choose &apos;Import pubky&apos; option <br />
    3. Tap &apos;Scan QR to import&apos; <br />
    4. Scan this QR code to import your recovery phrase
  </>
);

const EXPORT_DESCRIPTION = (
  <>
    1. Open Pubky Ring, tap &apos;Add pubky&apos; <br />
    2. Choose the &apos;Import pubky&apos; option
    <br />
    3. Scan this QR to import
  </>
);

const MOBILE_EXPORT_DESCRIPTION =
  'Tap the button below to export your pubky and import it into Pubky Ring for a safer and easier sign-in experience.';

const MOBILE_IMPORT_DESCRIPTION =
  'Tap the button below to import your recovery phrase into Pubky Ring for a safer and easier sign-in experience.';

export function DialogBackupExport({ mnemonic, children }: DialogBackupExportProps) {
  const qrValue = mnemonic ? Libs.generatePubkyRingDeeplink(mnemonic) : Config.PUBKY_CORE_URL;
  const descriptionContent = mnemonic ? IMPORT_DESCRIPTION : EXPORT_DESCRIPTION;
  const mobileDescription = mnemonic ? MOBILE_IMPORT_DESCRIPTION : MOBILE_EXPORT_DESCRIPTION;
  const dialogTitle = mnemonic ? 'Import recovery phrase' : 'Pubky Ring export';

  const handleMobileButtonClick = () => {
    if (typeof window !== 'undefined') {
      const url = mnemonic ? Libs.generatePubkyRingDeeplink(mnemonic) : Config.PUBKY_CORE_URL;
      window.open(url, '_blank');
    }
  };

  return (
    <Atoms.Dialog>
      {children ? (
        <Atoms.DialogTrigger asChild>{children}</Atoms.DialogTrigger>
      ) : (
        <Atoms.DialogTrigger asChild>
          <Atoms.Button>Continue</Atoms.Button>
        </Atoms.DialogTrigger>
      )}
      <Atoms.DialogContent className="max-w-md" hiddenTitle={dialogTitle}>
        {/* Desktop version - hidden on mobile, shown on desktop */}
        <div className="hidden lg:flex flex-col gap-6">
          <Atoms.DialogHeader>
            <Atoms.DialogTitle>{dialogTitle}</Atoms.DialogTitle>
            <Atoms.DialogDescription>{descriptionContent}</Atoms.DialogDescription>
          </Atoms.DialogHeader>

          <Atoms.Container className="justify-between items-start">
            <Atoms.Container className="flex-row gap-8">
              <Atoms.Container>
                <Image src="/images/pubky-ring-phone.png" alt="App preview" width={250} height={430} />
              </Atoms.Container>

              <Atoms.Container className="gap-6">
                <Atoms.Container className="mx-0 bg-foreground rounded-lg p-[9px] w-[192px] h-[192px] items-center">
                  <QRCodeSVG value={qrValue} size={174} />
                </Atoms.Container>
                <Atoms.Container className="gap-4">
                  <Atoms.Link href={Config.APP_STORE_URL} target="_blank">
                    <Image src="/images/badge-apple.png" alt="App Store" width={120} height={40} />
                  </Atoms.Link>
                  <Atoms.Link href={Config.PLAY_STORE_URL} target="_blank">
                    <Image src="/images/badge-android.png" alt="Google Play" width={135} height={40} />
                  </Atoms.Link>
                </Atoms.Container>
              </Atoms.Container>
            </Atoms.Container>
          </Atoms.Container>
        </div>

        {/* Mobile version - shown on mobile, hidden on desktop */}
        <div className="flex lg:hidden flex-col gap-6">
          <Atoms.DialogHeader>
            <Atoms.DialogTitle>{dialogTitle}</Atoms.DialogTitle>
            <Atoms.DialogDescription>{mobileDescription}</Atoms.DialogDescription>
          </Atoms.DialogHeader>

          <Atoms.Container className="gap-3">
            <Atoms.Button onClick={handleMobileButtonClick} className="w-full">
              Import to Pubky Ring
            </Atoms.Button>
          </Atoms.Container>
        </div>
      </Atoms.DialogContent>
    </Atoms.Dialog>
  );
}
