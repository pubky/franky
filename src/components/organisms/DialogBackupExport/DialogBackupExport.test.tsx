import { describe, it, expect, vi, beforeEach, afterEach } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import { DialogBackupExport } from './DialogBackupExport';

// Mock Next.js Image component
vi.mock('next/image', () => ({
  default: ({
    src,
    alt,
    width,
    height,
    className,
  }: {
    src: string;
    alt: string;
    width: number;
    height: number;
    className?: string;
  }) => <img src={src} alt={alt} width={width} height={height} className={className} data-testid="next-image" />,
}));

// Mock QRCodeSVG component
vi.mock('qrcode.react', () => ({
  QRCodeSVG: ({ value, size }: { value: string; size: number }) => (
    <div data-testid="qr-code" data-value={value} data-size={size}>
      QR Code
    </div>
  ),
}));

// Mock @/libs - use actual implementations (no icon stubs)
vi.mock('@/libs', async (importOriginal) => {
  const actual = await importOriginal<typeof import('@/libs')>();
  return { ...actual };
});

// Mock window.open
const mockWindowOpen = vi.fn();
const originalOpen = window.open;

// Mock atoms
vi.mock('@/atoms', () => ({
  Dialog: ({ children }: { children: React.ReactNode }) => <div data-testid="dialog">{children}</div>,
  DialogTrigger: ({ children, asChild }: { children: React.ReactNode; asChild?: boolean }) => (
    <div data-testid="dialog-trigger" data-as-child={asChild}>
      {children}
    </div>
  ),
  DialogContent: ({ children, className }: { children: React.ReactNode; className?: string }) => (
    <div data-testid="dialog-content" className={className}>
      {children}
    </div>
  ),
  DialogHeader: ({ children, className }: { children: React.ReactNode; className?: string }) => (
    <div data-testid="dialog-header" className={className}>
      {children}
    </div>
  ),
  DialogTitle: ({ children, className }: { children: React.ReactNode; className?: string }) => (
    <h2 data-testid="dialog-title" className={className}>
      {children}
    </h2>
  ),
  DialogDescription: ({ children, className }: { children: React.ReactNode; className?: string }) => (
    <p data-testid="dialog-description" className={className}>
      {children}
    </p>
  ),
  Button: ({
    children,
    variant,
    size,
    className,
    onClick,
    ...props
  }: {
    children: React.ReactNode;
    variant?: string;
    size?: string;
    className?: string;
    onClick?: () => void;
    [key: string]: unknown;
  }) => (
    <button
      data-testid={variant ? `button-${variant}` : 'button'}
      data-variant={variant}
      data-size={size}
      className={className}
      onClick={onClick}
      {...props}
    >
      {children}
    </button>
  ),
  Link: ({
    children,
    href,
    target,
    className,
  }: {
    children: React.ReactNode;
    href: string;
    target?: string;
    className?: string;
  }) => (
    <a data-testid="link" href={href} target={target} className={className}>
      {children}
    </a>
  ),
  Container: ({ children, className }: { children: React.ReactNode; className?: string }) => (
    <div data-testid="container" className={className}>
      {children}
    </div>
  ),
}));

describe('DialogBackupExport', () => {
  beforeEach(() => {
    Object.defineProperty(window, 'open', {
      configurable: true,
      value: mockWindowOpen,
    });
    vi.clearAllMocks();
  });

  afterEach(() => {
    Object.defineProperty(window, 'open', {
      configurable: true,
      value: originalOpen,
    });
  });

  it('renders with default props', () => {
    render(<DialogBackupExport />);

    const dialog = screen.getByTestId('dialog');
    const trigger = screen.getByTestId('dialog-trigger');
    const content = screen.getByTestId('dialog-content');

    expect(dialog).toBeInTheDocument();
    expect(trigger).toBeInTheDocument();
    expect(content).toBeInTheDocument();
  });

  it('handles click events on trigger button', () => {
    render(<DialogBackupExport />);

    const buttons = screen.getAllByTestId('button');
    const triggerButton = buttons.find((btn) => btn.textContent === 'Continue');

    expect(triggerButton).toBeInTheDocument();
    fireEvent.click(triggerButton!);
  });

  it('renders desktop version with correct classes', () => {
    const { container } = render(<DialogBackupExport />);

    const desktopVersion = container.querySelector('.hidden.lg\\:flex');
    expect(desktopVersion).toBeInTheDocument();
  });

  it('renders mobile version with correct classes', () => {
    const { container } = render(<DialogBackupExport />);

    const mobileVersion = container.querySelector('.flex.lg\\:hidden');
    expect(mobileVersion).toBeInTheDocument();
  });

  describe('Desktop version', () => {
    it('contains proper desktop content structure', () => {
      render(<DialogBackupExport />);

      // Check that all main elements are present
      const titles = screen.getAllByText('Pubky Ring export');
      expect(titles.length).toBeGreaterThan(0);

      expect(screen.getByText(/Open Pubky Ring, tap 'Add pubky'/)).toBeInTheDocument();

      // Desktop version should have QR code
      const qrCodes = screen.getAllByTestId('qr-code');
      expect(qrCodes.length).toBeGreaterThan(0);
      const desktopQrCode = qrCodes[0];
      expect(desktopQrCode).toBeInTheDocument();
      expect(desktopQrCode).toHaveAttribute('data-size', '174');

      expect(screen.getByAltText('App preview')).toBeInTheDocument();
      expect(screen.getByAltText('App Store')).toBeInTheDocument();
      expect(screen.getByAltText('Google Play')).toBeInTheDocument();
    });

    it('has correct QR code container padding', () => {
      const { container } = render(<DialogBackupExport />);

      const qrContainer = container.querySelector('[class*="p-[9px]"]');
      expect(qrContainer).toBeInTheDocument();
    });

    it('has correct container alignment (items-start)', () => {
      render(<DialogBackupExport />);

      const containers = screen.getAllByTestId('container');
      const mainContainer = containers.find(
        (container) => container.className.includes('justify-between') && container.className.includes('items-start'),
      );
      expect(mainContainer).toBeInTheDocument();
    });

    it('has correct gap between phone and QR sections (gap-8)', () => {
      render(<DialogBackupExport />);

      const containers = screen.getAllByTestId('container');
      const flexRowContainer = containers.find(
        (container) => container.className.includes('flex-row') && container.className.includes('gap-8'),
      );
      expect(flexRowContainer).toBeInTheDocument();
    });

    it('has correct max-width (max-w-md)', () => {
      render(<DialogBackupExport />);

      const content = screen.getByTestId('dialog-content');
      expect(content).toHaveClass('max-w-md');
    });
  });

  describe('Mobile version', () => {
    it('renders mobile description text', () => {
      render(<DialogBackupExport />);

      expect(
        screen.getByText(/Tap the button below to export your pubky and import it into Pubky Ring/),
      ).toBeInTheDocument();
    });

    it('renders mobile button', () => {
      render(<DialogBackupExport />);

      const buttons = screen.getAllByTestId('button');
      const mobileButton = buttons.find((btn) => btn.textContent === 'Import to Pubky Ring');
      expect(mobileButton).toBeInTheDocument();
      expect(mobileButton).toHaveClass('w-full');
    });

    it('opens deeplink when mobile button is clicked', () => {
      render(<DialogBackupExport />);

      const buttons = screen.getAllByTestId('button');
      const mobileButton = buttons.find((btn) => btn.textContent === 'Import to Pubky Ring');

      fireEvent.click(mobileButton!);

      expect(mockWindowOpen).toHaveBeenCalled();
    });

    it('does not render QR code or store badges in mobile version', () => {
      const { container } = render(<DialogBackupExport />);

      // QR codes should only exist in desktop version
      const qrCodes = screen.getAllByTestId('qr-code');
      expect(qrCodes.length).toBeGreaterThan(0);

      // Store badges should only exist in desktop version
      const links = screen.getAllByTestId('link');
      expect(links.length).toBeGreaterThan(0);

      // Mobile version should not have QR codes or links
      const mobileContainer = container.querySelector('.flex.lg\\:hidden');
      const mobileQrCode = mobileContainer?.querySelector('[data-testid="qr-code"]');
      const mobileLinks = mobileContainer?.querySelectorAll('[data-testid="link"]');
      expect(mobileQrCode).not.toBeInTheDocument();
      expect(mobileLinks?.length || 0).toBe(0);
    });
  });

  describe('with mnemonic prop', () => {
    it('renders with mnemonic-specific title', () => {
      const testMnemonic = 'wood fox silver drive march fee palace flame earn door case almost';
      render(<DialogBackupExport mnemonic={testMnemonic} />);

      const titles = screen.getAllByTestId('dialog-title');
      titles.forEach((title) => {
        expect(title).toHaveTextContent('Import recovery phrase');
      });
    });

    it('renders mobile import description for mnemonic', () => {
      const testMnemonic = 'wood fox silver drive march fee palace flame earn door case almost';
      render(<DialogBackupExport mnemonic={testMnemonic} />);

      expect(
        screen.getByText(/Tap the button below to import your recovery phrase into Pubky Ring/),
      ).toBeInTheDocument();
    });

    it('opens correct deeplink when mobile button is clicked with mnemonic', () => {
      const testMnemonic = 'wood fox silver drive march fee palace flame earn door case almost';
      render(<DialogBackupExport mnemonic={testMnemonic} />);

      const buttons = screen.getAllByTestId('button');
      const mobileButton = buttons.find((btn) => btn.textContent === 'Import to Pubky Ring');

      fireEvent.click(mobileButton!);

      expect(mockWindowOpen).toHaveBeenCalled();
    });

    it('renders desktop version with correct QR code size for mnemonic', () => {
      const testMnemonic = 'wood fox silver drive march fee palace flame earn door case almost';
      render(<DialogBackupExport mnemonic={testMnemonic} />);

      const qrCodes = screen.getAllByTestId('qr-code');
      expect(qrCodes.length).toBeGreaterThan(0);
      const desktopQrCode = qrCodes[0];
      expect(desktopQrCode).toHaveAttribute('data-size', '174');
    });
  });
});

describe('DialogBackupExport - Snapshots', () => {
  beforeEach(() => {
    Object.defineProperty(window, 'open', {
      configurable: true,
      value: mockWindowOpen,
    });
  });

  afterEach(() => {
    Object.defineProperty(window, 'open', {
      configurable: true,
      value: originalOpen,
    });
  });

  it('matches snapshot for default DialogBackupExport', () => {
    const { container } = render(<DialogBackupExport />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot for DialogBackupExport with mnemonic', () => {
    const testMnemonic = 'wood fox silver drive march fee palace flame earn door case almost';
    const { container } = render(<DialogBackupExport mnemonic={testMnemonic} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot for DialogBackupExport with special mnemonic', () => {
    const specialMnemonic = 'test phrase with spaces & symbols!';
    const { container } = render(<DialogBackupExport mnemonic={specialMnemonic} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
