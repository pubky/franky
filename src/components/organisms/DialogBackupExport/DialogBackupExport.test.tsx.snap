// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DialogBackupExport - Snapshots > matches snapshot for DialogBackupExport with mnemonic 1`] = `
<div
  data-testid="dialog"
>
  <div
    data-as-child="true"
    data-testid="dialog-trigger"
  >
    <button
      data-testid="button"
    >
      Continue
    </button>
  </div>
  <div
    class="max-w-md"
    data-testid="dialog-content"
  >
    <div
      class="hidden lg:flex flex-col gap-6"
    >
      <div
        data-testid="dialog-header"
      >
        <h2
          data-testid="dialog-title"
        >
          Import recovery phrase
        </h2>
        <p
          data-testid="dialog-description"
        >
          1. Open Pubky Ring, tap 'Add pubky' 
          <br />
          2. Choose 'Import pubky' option 
          <br />
          3. Tap 'Scan QR to import' 
          <br />
          4. Scan this QR code to import your recovery phrase
        </p>
      </div>
      <div
        class="justify-between items-start"
        data-testid="container"
      >
        <div
          class="flex-row gap-8"
          data-testid="container"
        >
          <div
            data-testid="container"
          >
            <img
              alt="App preview"
              data-testid="next-image"
              height="430"
              src="/images/pubky-ring-phone.png"
              width="250"
            />
          </div>
          <div
            class="gap-6"
            data-testid="container"
          >
            <div
              class="mx-0 bg-foreground rounded-lg p-[9px] w-[192px] h-[192px] items-center"
              data-testid="container"
            >
              <div
                data-size="174"
                data-testid="qr-code"
                data-value="pubkyring://wood%20fox%20silver%20drive%20march%20fee%20palace%20flame%20earn%20door%20case%20almost"
              >
                QR Code
              </div>
            </div>
            <div
              class="gap-4"
              data-testid="container"
            >
              <a
                data-testid="link"
                href="https://apps.apple.com/app/pubky-ring/id6739356756"
                target="_blank"
              >
                <img
                  alt="App Store"
                  data-testid="next-image"
                  height="40"
                  src="/images/badge-apple.png"
                  width="120"
                />
              </a>
              <a
                data-testid="link"
                href="https://play.google.com/store/apps/details?id=to.pubky.ring&pcampaignid=web_share"
                target="_blank"
              >
                <img
                  alt="Google Play"
                  data-testid="next-image"
                  height="40"
                  src="/images/badge-android.png"
                  width="135"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex lg:hidden flex-col gap-6"
    >
      <div
        data-testid="dialog-header"
      >
        <h2
          data-testid="dialog-title"
        >
          Import recovery phrase
        </h2>
        <p
          data-testid="dialog-description"
        >
          Tap the button below to import your recovery phrase into Pubky Ring for a safer and easier sign-in experience.
        </p>
      </div>
      <div
        class="gap-3"
        data-testid="container"
      >
        <button
          class="w-full"
          data-testid="button"
        >
          Import to Pubky Ring
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`DialogBackupExport - Snapshots > matches snapshot for DialogBackupExport with special mnemonic 1`] = `
<div
  data-testid="dialog"
>
  <div
    data-as-child="true"
    data-testid="dialog-trigger"
  >
    <button
      data-testid="button"
    >
      Continue
    </button>
  </div>
  <div
    class="max-w-md"
    data-testid="dialog-content"
  >
    <div
      class="hidden lg:flex flex-col gap-6"
    >
      <div
        data-testid="dialog-header"
      >
        <h2
          data-testid="dialog-title"
        >
          Import recovery phrase
        </h2>
        <p
          data-testid="dialog-description"
        >
          1. Open Pubky Ring, tap 'Add pubky' 
          <br />
          2. Choose 'Import pubky' option 
          <br />
          3. Tap 'Scan QR to import' 
          <br />
          4. Scan this QR code to import your recovery phrase
        </p>
      </div>
      <div
        class="justify-between items-start"
        data-testid="container"
      >
        <div
          class="flex-row gap-8"
          data-testid="container"
        >
          <div
            data-testid="container"
          >
            <img
              alt="App preview"
              data-testid="next-image"
              height="430"
              src="/images/pubky-ring-phone.png"
              width="250"
            />
          </div>
          <div
            class="gap-6"
            data-testid="container"
          >
            <div
              class="mx-0 bg-foreground rounded-lg p-[9px] w-[192px] h-[192px] items-center"
              data-testid="container"
            >
              <div
                data-size="174"
                data-testid="qr-code"
                data-value="pubkyring://test%20phrase%20with%20spaces%20%26%20symbols!"
              >
                QR Code
              </div>
            </div>
            <div
              class="gap-4"
              data-testid="container"
            >
              <a
                data-testid="link"
                href="https://apps.apple.com/app/pubky-ring/id6739356756"
                target="_blank"
              >
                <img
                  alt="App Store"
                  data-testid="next-image"
                  height="40"
                  src="/images/badge-apple.png"
                  width="120"
                />
              </a>
              <a
                data-testid="link"
                href="https://play.google.com/store/apps/details?id=to.pubky.ring&pcampaignid=web_share"
                target="_blank"
              >
                <img
                  alt="Google Play"
                  data-testid="next-image"
                  height="40"
                  src="/images/badge-android.png"
                  width="135"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex lg:hidden flex-col gap-6"
    >
      <div
        data-testid="dialog-header"
      >
        <h2
          data-testid="dialog-title"
        >
          Import recovery phrase
        </h2>
        <p
          data-testid="dialog-description"
        >
          Tap the button below to import your recovery phrase into Pubky Ring for a safer and easier sign-in experience.
        </p>
      </div>
      <div
        class="gap-3"
        data-testid="container"
      >
        <button
          class="w-full"
          data-testid="button"
        >
          Import to Pubky Ring
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`DialogBackupExport - Snapshots > matches snapshot for default DialogBackupExport 1`] = `
<div
  data-testid="dialog"
>
  <div
    data-as-child="true"
    data-testid="dialog-trigger"
  >
    <button
      data-testid="button"
    >
      Continue
    </button>
  </div>
  <div
    class="max-w-md"
    data-testid="dialog-content"
  >
    <div
      class="hidden lg:flex flex-col gap-6"
    >
      <div
        data-testid="dialog-header"
      >
        <h2
          data-testid="dialog-title"
        >
          Pubky Ring export
        </h2>
        <p
          data-testid="dialog-description"
        >
          1. Open Pubky Ring, tap 'Add pubky' 
          <br />
          2. Choose the 'Import pubky' option
          <br />
          3. Scan this QR to import
        </p>
      </div>
      <div
        class="justify-between items-start"
        data-testid="container"
      >
        <div
          class="flex-row gap-8"
          data-testid="container"
        >
          <div
            data-testid="container"
          >
            <img
              alt="App preview"
              data-testid="next-image"
              height="430"
              src="/images/pubky-ring-phone.png"
              width="250"
            />
          </div>
          <div
            class="gap-6"
            data-testid="container"
          >
            <div
              class="mx-0 bg-foreground rounded-lg p-[9px] w-[192px] h-[192px] items-center"
              data-testid="container"
            >
              <div
                data-size="174"
                data-testid="qr-code"
                data-value="https://pubky.org"
              >
                QR Code
              </div>
            </div>
            <div
              class="gap-4"
              data-testid="container"
            >
              <a
                data-testid="link"
                href="https://apps.apple.com/app/pubky-ring/id6739356756"
                target="_blank"
              >
                <img
                  alt="App Store"
                  data-testid="next-image"
                  height="40"
                  src="/images/badge-apple.png"
                  width="120"
                />
              </a>
              <a
                data-testid="link"
                href="https://play.google.com/store/apps/details?id=to.pubky.ring&pcampaignid=web_share"
                target="_blank"
              >
                <img
                  alt="Google Play"
                  data-testid="next-image"
                  height="40"
                  src="/images/badge-android.png"
                  width="135"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex lg:hidden flex-col gap-6"
    >
      <div
        data-testid="dialog-header"
      >
        <h2
          data-testid="dialog-title"
        >
          Pubky Ring export
        </h2>
        <p
          data-testid="dialog-description"
        >
          Tap the button below to export your pubky and import it into Pubky Ring for a safer and easier sign-in experience.
        </p>
      </div>
      <div
        class="gap-3"
        data-testid="container"
      >
        <button
          class="w-full"
          data-testid="button"
        >
          Import to Pubky Ring
        </button>
      </div>
    </div>
  </div>
</div>
`;
